DSpace/DSpace
# Features endpoint: Download/RequestACopy features

## References
* Fixes https://github.com/DSpace/DSpace/issues/2943

## Description
We've added REST Authorization Features: 
* canDownload: Can the current Bitstream or Bitstreams in an item be downloaded
* canRequestACopy: Can a copy of the current Bitstream be requested

## Instructions for Reviewers

List of changes in this PR:
* Added the DownloadFeature and RequestCopyFeature features for the Bitstream/Item and Bitstream objects respectively
* Provided tests for these new features

To test this PR:
* Navigate to the {dspace.url}/server/api/authz/authorizations/search/object?uri={dspace.url}/server/api/core/bitstreams/{bitstreamID}&embed=feature whilst being logged in as admin and verify that you can see the DownloadFeature
* Navigate to the {dspace.url}/server/api/authz/authorizations/search/object?uri={dspace.url}/server/api/core/bitstreams/{bitstreamID}&embed=feature with the Bitstream being a private Bitstream and whilst not being logged in. verify that you can see the RequestACopyFeature
* Navigate to the {dspace.url}/server/api/authz/authorizations/search/object?uri={dspace.url}/server/api/core/items/{itemID}&embed=feature with the Item being an Item that contains a private Bitstream and whilst not being logged in. verify that you can see the RequestACopyFeature

## Checklist
